src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,81 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { useState } from 'react';
+import { FilterOptions } from '../../types/Filter';
+
+type Props = {
+  onFilterChange: (filter: FilterOptions) => void;
+  onSearchChange: (query: string) => void;
+  onClose: () => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onFilterChange,
+  onSearchChange,
+  onClose,
+}) => {
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.All);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newValue = event.target.value;
+
+    if (Object.values(FilterOptions).includes(newValue as FilterOptions)) {
+      const newFilter = newValue as FilterOptions;
```

Comment:
The type casting of `event.target.value` to `FilterOptions` assumes it will always be valid. Ensure that the value is a valid `FilterOptions` to prevent runtime errors. Consider adding a type guard or validation logic to handle invalid cases.